src/getTables.ts

```diff
@@ -46,9 +46,9 @@ async function getTables(
             );
         } else {
             // only include the tables from the options that actually exist in the db
-            tables = tables.filter(
-                t => restrictedTables.indexOf(t.name) !== -1,
-            );
+            tables = restrictedTables
+                .map(tableName => actualTables.find(t => t.name === tableName)) // keeping the order of the passed-in whitelist
+                .filter(t => Boolean(t)) as Array<Table>; // filter out non-existing tables
```

Comment:
You should avoid using a type cast unless it's _absolutely_ necessary.
Also you should avoid using the `Boolean` constructor because it does too much magic in terms of truthy/falsey coercion.

For `filter`, you can annotate with a type predicate
```suggestion
                // filter out non-existing tables
                .filter((t): t is Table => t != null);
```

Reply:
this will give you the lint error: `Expected '!==' and instead saw '!='  eqeqeq` ...